doc/security_considerations.md

```diff
@@ -1,15 +1,32 @@
+<!---
+Copyright 2018-2021 Crown Copyright
 
-Where this gets trickier is when the user is authenticating with one of the palisade services or one of the data service. The main reason this gets tricky is because we expect Palisade to be used to access data in a distributed way over a cluster.
-Therefore the client authentication token is only going to be available on the machine that the client is running the client code on and not necessarily available at the point at which the distributed code is requesting the data from the data services.
-That is why we have to register the data request with the palisade service first in a non-distributed way where you can provide a personal token/certificate to be validated to ensure authentication.
-Then the palisade service can then provide a token to be passed to the data service to authenticate that the same already authenticated user is requesting access to the desired resource.
-The data service validates the token with a palisade service and uses it to get the required trusted details from a cache service that is shared by all instances of the palisade service.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
 
-This is to satisfy Hadoop's security architecture. In environments where users run as themselves (as opposed to everybody running as the Hadoop processing user), we can do client-auth at on connection to the PalisadeService and again at the connection to DataService.
-The underlying mechanism still works identically, but in addition we can also validate that a user is using the Token assigned to them by the PalisadeService and detect if that Token is being used by a different user (spoofing).  This validation check can not be done with Hadoop.
+  http://www.apache.org/licenses/LICENSE-2.0
 
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+--->
+
+# Security considerations
+
+During the design of Palisade, we spent some time thinking about how we would be able to authenticate who the user is (the user that is requesting access to the data).
+The security boundaries for the recommended K8s deployment of Palisade are at the point of the client connecting to the Palisade, Filtered-Resource or Data Services (any service exposed by some ingress).
+This can mostly be handled by the User Service authenticating the original request and credentials from the Palsiade Service, with all others using the supplied token generated by the Palisade Service.
```

Comment:
```suggestion
This can mostly be handled by the User Service authenticating the original request and credentials from the Palisade Service, with all others using the supplied token generated by the Palisade Service.
```